App/Services/UserNotifier.cs

```diff
@@ -0,0 +1,29 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Windows.AppNotifications;
+using Microsoft.Windows.AppNotifications.Builder;
+
+namespace Coder.Desktop.App.Services;
+
+public interface IUserNotifier : IAsyncDisposable
+{
+    public Task ShowErrorNotification(string title, string message);
+}
+
+public class UserNotifier : IUserNotifier
+{
+    private readonly AppNotificationManager _notificationManager = AppNotificationManager.Default;
+
+    public ValueTask DisposeAsync()
+    {
+        return ValueTask.CompletedTask;
+    }
+
+    public Task ShowErrorNotification(string title, string message)
```

Comment:
This could just be `ShowNotification`, the function itself doesn't seem to be only relevant for errors. I imagine we could use it for other notifications like workspace lifetime notifications in the future

Reply:
I kinda want to keep the information that this is an "error" in case we want to refactor this class to use, say, a dialog pop-up instead, or decide to style the notifications further to make errors stand out from other notifications.